r/tests/testthat/test-dplyr-funcs-datetime.R

```diff
@@ -1942,3 +1925,206 @@ test_that("lubridate's fast_strptime", {
+        dmy_h_dttm   = parse_date_time(dmy_h_string, orders = "dmy_H"),
+        mdy_hms_dttm = parse_date_time(mdy_hms_string, orders = "mdy_HMS"),
+        mdy_hm_dttm  = parse_date_time(mdy_hm_string, orders = "mdy_HM"),
+        mdy_h_dttm   = parse_date_time(mdy_h_string, orders = "mdy_H"),
+        ydm_hms_dttm = parse_date_time(ydm_hms_string, orders = "ydm_HMS"),
+        ydm_hm_dttm  = parse_date_time(ydm_hm_string, orders = "ydmHM"),
+        ydm_h_dttm   = parse_date_time(ydm_h_string, orders = "ydmH")
+      ) %>%
+      collect(),
+    test_dates_times
+  )
+
+  compare_dplyr_binding(
+    .input %>%
+      mutate(
+        ymd_hms_dttm =
+          parse_date_time(ymd_hms_string, orders = "ymd_HMS", tz = "Pacific/Marquesas"),
+        ymd_hm_dttm =
+          parse_date_time(ymd_hm_string, orders = "ymd_HM", tz = "Pacific/Marquesas"),
+        ymd_h_dttm =
+          parse_date_time(ymd_h_string, orders = "ymd_H", tz = "Pacific/Marquesas"),
+        dmy_hms_dttm =
+          parse_date_time(dmy_hms_string, orders = "dmy_HMS", tz = "Pacific/Marquesas"),
+        dmy_hm_dttm =
+          parse_date_time(dmy_hm_string, orders = "dmy_HM", tz = "Pacific/Marquesas"),
+        dmy_h_dttm =
+          parse_date_time(dmy_h_string, orders = "dmy_H", tz = "Pacific/Marquesas"),
+        mdy_hms_dttm =
+          parse_date_time(mdy_hms_string, orders = "mdy_HMS", tz = "Pacific/Marquesas"),
+        mdy_hm_dttm =
+          parse_date_time(mdy_hm_string, orders = "mdy_HM", tz = "Pacific/Marquesas"),
+        mdy_h_dttm =
+          parse_date_time(mdy_h_string, orders = "mdy_H", tz = "Pacific/Marquesas"),
+        ydm_hms_dttm =
+          parse_date_time(ydm_hms_string, orders = "ydm_HMS", tz = "Pacific/Marquesas"),
+        ydm_hm_dttm =
+          parse_date_time(ydm_hm_string, orders = "ydm_HM", tz = "Pacific/Marquesas"),
+        ydm_h_dttm =
+          parse_date_time(ydm_h_string, orders = "ydm_H", tz = "Pacific/Marquesas")
+      ) %>%
+      collect(),
+    test_dates_times
+  )
+
+  # test truncated formats
+  compare_dplyr_binding(
+    .input %>%
+      mutate(
+        dttm =
+          parse_date_time(
+            truncated_ymd_string,
+            orders = "ymd_HMS",
+            truncated = 3
+          )
+      ) %>%
+      collect(),
+    tibble(
+      truncated_ymd_string =
+        c(
+          "2022-05-19 13:46:51",
+          "2022-05-18 13:46",
+          "2022-05-17 13",
+          "2022-05-16"
+        )
+    )
+  )
+
+  # we need expect_warning twice as both the arrow pipeline (because quiet =
+  # FALSE is not supported) and the fallback dplyr/lubridate one throw
+  # warnings (the lubridate one because quiet is FALSE)
+  expect_warning(
+    expect_warning(
+      tibble(x = c("2022-05-19 13:46:51")) %>%
+        arrow_table() %>%
+        mutate(
+          x_dttm = parse_date_time(x, orders = "dmy_HMS", quiet = FALSE)
+        ) %>%
+        collect(),
+      "`quiet = FALSE` not supported in Arrow"
+    )
+  )
```

Comment:
Should we assert the second warning here?

Reply:
Done